pkg/controller/volume/persistentvolume/util/util.go

```diff
@@ -94,34 +94,24 @@ func IsDelayBindingMode(claim *v1.PersistentVolumeClaim, classLister storagelist
 func GetBindVolumeToClaim(volume *v1.PersistentVolume, claim *v1.PersistentVolumeClaim) (*v1.PersistentVolume, bool, error) {
 	dirty := false
 
-	// Check if the volume was already bound (either by user or by controller)
-	shouldSetBoundByController := false
-	if !IsVolumeBoundToClaim(volume, claim) {
-		shouldSetBoundByController = true
-	}
-
 	// The volume from method args can be pointing to watcher cache. We must not
 	// modify these, therefore create a copy.
 	volumeClone := volume.DeepCopy()
 
-	// Bind the volume to the claim if it is not bound yet
-	if volume.Spec.ClaimRef == nil ||
-		volume.Spec.ClaimRef.Name != claim.Name ||
-		volume.Spec.ClaimRef.Namespace != claim.Namespace ||
-		volume.Spec.ClaimRef.UID != claim.UID {
-
+	// Check if the volume was already bound (either by user or by controller)
```

Comment:
Given the code of IsVolumeBoundToClaim, this refactor seems valid

```go
func IsVolumeBoundToClaim(volume *v1.PersistentVolume, claim *v1.PersistentVolumeClaim) bool {
	if volume.Spec.ClaimRef == nil {
		return false
	}
	if claim.Name != volume.Spec.ClaimRef.Name || claim.Namespace != volume.Spec.ClaimRef.Namespace {
		return false
	}
	if volume.Spec.ClaimRef.UID != "" && claim.UID != volume.Spec.ClaimRef.UID {
		return false
	}
	return true
}
```

Reply:
I didn't know that pre-bound volume and that it has empty `volume.Spec.ClaimRef.UID`. So changes of this PR became only removing an unnecessary intermediate variable and adding more comments about pre-bound volume.